src/components/SummaryNameMapModal.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+    Select,
+    Button,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalBody,
+    ModalFooter,
+    FormControl,
+    FormLabel,
+    ModalOverlay,
+    VStack
+} from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import trpc from "../trpc";
+import ModalWithBackdrop from 'components/ModalWithBackdrop';
+import ReactMarkdown from 'react-markdown';
+import { MinUser } from 'shared/User';
+import { SummaryNameMap } from 'shared/Summary';
+
+export function SummaryNameMapModal({ nameMap, groupUsers, onClose }: {
+    nameMap: SummaryNameMap | undefined
+    groupUsers: MinUser[]
+    onClose: () => void,
+}) {
+    // UserMap = { [handlebar] : { name: [userName], id: [userId] }}
+    // May update summaryNameMap API to output UserMap instead
+    const [updatedUserMap, setUpdatedUserMap] = useState<{ [key: string]: SummaryNameMap }>({});
+
+    useEffect(() => {
```

Comment:
No need for useEffect. the computation isn't that expensive.

Reply:
useEffect is needed for the page to render, otherwise will return blank page